### DISCUSSION TITLE
Slowing down the waveshare Jetbot

### DISCUSSION THREAD

**Discussion by DingoOz** [NVIDIA-AI-IOT/jetbot#377] [General]
From another discussion thread it is clear that the Waveshare board is driving its motors with ~12V from the onboard batteries.

This results in very jerky and rapid acceleration of the motors whenever they are engaged.

Has anyone delved into the code to slow down the Waveshare motors to be similar in performance to the other Jetbots?

I am thinking of three ways to slow down the motors:
* hardware change to drop the voltage to 5V - 6V (this would waste current, loses torque, adds heat and generally be least desirable)
* slow the current PWM duty cycle
* slow the current PWM duty cycle but also add a the "breaking" H-bridge commands to the "off" duty cycle.

Has anyone already gone down this path with the Waveshare bots?

**Comment by chitoku** [comment]
Hi DingoOz,

I just tried your mehtod-1, by adding a [DC/DC converter](https://www.amazon.com/gp/product/B01MQGMOKI/) to interject 5V to the motor driver as VM.
The result was good. I think I'm getting much lower, controllable speed, seemingly very close to the original DIY JetBot.

![IMG_1472](https://user-images.githubusercontent.com/262161/116967207-a59b6b80-ac66-11eb-8ec4-47eb35dcfefb.jpg)
![IMG_1471](https://user-images.githubusercontent.com/262161/116967218-a9c78900-ac66-11eb-862b-299bf1f92ab8.jpg)
![IMG_1473](https://user-images.githubusercontent.com/262161/116967210-a7fdc580-ac66-11eb-8cd2-bc870f7f988b.jpg)

Just to explain what's done on a schematic.

<img width="833" alt="2021-05-03_23h51_08" src="https://user-images.githubusercontent.com/262161/116969276-da112680-ac6a-11eb-8978-8593cd624b42.png">

**Reply by DingoOz** [reply]
I very much like what you have done there. Smart set up!

**Reply by DingoOz** [reply]
![5v_mod](https://user-images.githubusercontent.com/1388693/125947156-99e05bbc-0bc5-4b56-a8af-f439206efa25.jpg)

I copied your approach. I accidently broke off pin VM1 from the TB6612 chip, but I was able to continue by solder VM2 and VM3 to 5V still.

I used a RECOM 78E05-1 to convert the 12v to 5v. It is a DC to DC convertor in a 7805 package. 

I felt confident from your post that I did not need the switch.

...

The whole thing was a complete success!

With the motors now receiving 5V and not 12V the entire robot is far more controllable.

**Reply by DingoOz** [reply]
https://odysee.com/@Robotics:d/5v_jetbot:8

This is a little video showing how smooth and predictable it is using 5V.

**Reply by JHPHELAN** [reply]
You can also change the motor speed in the software:
`robot.left(speed=0.3)`
or when that speed parameter is previously defined as above, just
'robot.left(0.3)'
Change it to 0.1 or some other value to slow the robot

**Reply by brianlmerritt** [reply]
Did anyone try modifying the code in the waveshare version of the jetbot?  jetbot/motor.py looks like the right place to try.

**Reply by DingoOz** [reply]
> You can also change the motor speed in the software: `robot.left(speed=0.3)` or when that speed parameter is previously defined as above, just 'robot.left(0.3)' Change it to 0.1 or some other value to slow the robot

Did this approach work? What value worked well. I was not able to achieve a stable robot with any value as I found it would either be too high and overshoot the drive or too low and not have enough torque at the wheels to move the robot.
